src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipeController.cs

```diff
@@ -144,7 +140,78 @@ private void PollForTracingCommand(object state)
+        {
+            // Read the config file in once call.
+            byte[] configContents = File.ReadAllBytes(configFilePath);
+
+            // Convert the contents to a string.
+            string strConfigContents = Encoding.UTF8.GetString(configContents);
+
+            // Read all of the config options.
+            string outputPath = null;
+            string strProviderConfig = null;
+            string strCircularMB = null;
+
+            // Split the configuration entries by line.
+            string[] configEntries = strConfigContents.Split(ConfigFileLineDelimiters, StringSplitOptions.RemoveEmptyEntries);
+            foreach (string configEntry in configEntries)
+            {
+                //`Split the key and value by '='.
+                string[] entryComponents = configEntry.Split(ConfigEntryDelimiter);
+                if(entryComponents.Length == 2)
+                {
+                    string key = entryComponents[0];
+                    if (key.Equals(ConfigKey_Providers))
+                    {
+                        strProviderConfig = entryComponents[1];
+                    }
+                    else if (key.Equals(ConfigKey_OutputPath))
+                    {
+                        outputPath = entryComponents[1];
+                    }
+                    else if(key.Equals(ConfigKey_CircularMB))
+                    {
+                        strCircularMB = entryComponents[1];
+                    }
+                }
+            }
+
+            // Ensure that the output path is set.
+            if (string.IsNullOrEmpty(outputPath))
+            {
+                throw new ArgumentNullException(nameof(outputPath));
+            }
+
+            // Build the full path to the trace file.
+            string traceFileName = Assembly.GetEntryAssembly().GetName().Name + "." + Win32Native.GetCurrentProcessId() + NetPerfFileExtension;
+            string outputFile = Path.Combine(outputPath, traceFileName);
+
+            // Get the circular buffer size.
+            uint circularMB = DefaultCircularBufferMB;
+            if(!string.IsNullOrEmpty(strCircularMB))
+            {
+                circularMB = Convert.ToUInt32(strCircularMB);
+            }
+
+            // Initialize a new configuration object.
+            EventPipeConfiguration config = new EventPipeConfiguration(outputFile, circularMB);
+
+            // Set the provider configuration if specified.
+            if (!string.IsNullOrEmpty(strProviderConfig))
+            {
+                SetProviderConfiguration(strProviderConfig, config);
+            }
+            else
+            {
+                // If the provider configuration isn't specified, use the default.
+                config.EnableProviderRange(DefaultProviderConfiguration);
+            }
+
+            return config;
+        }
+
+        private static EventPipeConfiguration BuildConfigFromEnvironment()
```

Comment:
I would prefer that the File based and environment variable based modes are as similar as possible.   Thus we should not have EventPipeOutputFile, but EventPipeOuputPath, and it should be a directory (just like in the file case).   

I am ambivalent about specifying a directory and forcing the file to be named the same as the app.  (the less we mandate the better).  In the past, when I was also confronted with having to create potentially multiple output files, I chose asked the user to specify an OutputFilePrefix, which is a file path without a suffix, but it is expected that the system put the PROCESS_ID.netperf suffix after it.    What do you think of that?